packages/EasyErrorHandler/src/Providers/ErrorCodesFromEnumProvider.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyErrorHandler\Providers;
+
+use EonX\EasyErrorHandler\Annotations\AsErrorCodes;
+use EonX\EasyErrorHandler\DataTransferObjects\ErrorCodeDto;
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface;
+use PhpParser\Error;
+use PhpParser\Node\Stmt\Enum_;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+use ReflectionClass;
+
+final class ErrorCodesFromEnumProvider implements ErrorCodesProviderInterface
+{
+    private Parser $parser;
+
+    public function __construct(private readonly string $projectDir)
+    {
+        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+    }
+
+    public function provide(): array
+    {
+        $enums = $this->locateErrorCodesEnums();
+
+        if (\count($enums) === 0) {
+            return [];
+        }
+
+        $errorCodes = [];
+        foreach ($enums as $enum) {
+            $cases = $enum::cases();
+            foreach ($cases as $case) {
+                $splittedName = \preg_split(
+                    pattern: '/([A-Z\d][a-z\d]+)/u',
+                    subject: $case->name,
+                    flags: \PREG_SPLIT_DELIM_CAPTURE
+                );
+                if (\is_array($splittedName)) {
+                    $errorCodes[] = new ErrorCodeDto(
+                        originalName: $case->name,
+                        errorCode: $case->value,
+                        splittedName: \array_filter($splittedName, static fn ($value) => $value !== '')
+                    );
+                }
+            }
+        }
+
+        return $errorCodes;
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    public function locateErrorCodesEnums()
+    {
+        $directory = new \RecursiveDirectoryIterator($this->projectDir . '/src');
```

Comment:
In Laravel projects, we keep all the application code in the `app` directory. If we want to support Laravel projects, we should probably inject the `src`/`app` directory in the provider.

Reply:
My bad. Fixed